src/logging/BaseLogger.ts

```diff
@@ -15,23 +15,30 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import {LogItem} from "./LogItem.js";
-import {LogLevel, LogFilter} from "./LogFilter.js";
+import {LogItem} from "./LogItem";
+import {LogLevel, LogFilter} from "./LogFilter";
+import type {FilterCreator, LabelOrValues, LogCallback} from "./LogItem";
+import type {LogLevelOrNull} from "./LogFilter";
+// todo: should this import be here just for getting the type? should it instead be done when Platform.js --> Platform.ts?
+import type {Platform} from "../platform/web/Platform.js";
+
+
+export abstract class BaseLogger {
+    protected _openItems: Set<LogItem> = new Set();
+    protected _platform: Platform;
 
-export class BaseLogger {
     constructor({platform}) {
-        this._openItems = new Set();
         this._platform = platform;
     }
 
-    log(labelOrValues, logLevel = LogLevel.Info) {
+    log(labelOrValues: LabelOrValues, logLevel: LogLevel = LogLevel.Info) {
```

Comment:
Maybe, more correct, if you will add return type too (`:void`) like this: `log(labelOrValues: LabelOrValues, logLevel: LogLevel = LogLevel.Info): void` and in other places